src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs

```diff
@@ -314,6 +314,20 @@ private ISmartTokenFormatter CreateSmartTokenFormatter(OptionSet optionSet, IEnu
             return changes;
         }
 
+        private IEnumerable<IFormattingRule> GetTypingRules(Document document, SyntaxNode root, int position)
+        {
+            // make auto formatting on typing around missing brace case better
+            var token = root.FindToken(position);
+
+            if (token.Kind() == SyntaxKind.CloseBraceToken ||
+                token.Kind() == SyntaxKind.EndOfFileToken)
+            {
+                return SpecializedCollections.EmptyEnumerable<IFormattingRule>();
+            }
+
+            return SpecializedCollections.SingletonEnumerable(TypingFormattingRule.Instance);
```

Comment:
this code needs clear explanations about waht it is tring to do and why it is appropriate.  Why do we include the TypingFormattingRule.Instance if we're not on CloseBrace/EOF?

Reply:
I believe I copy and pasted existing comment as they are. if that is not sufficient, give me one, I will put that in the PR.